diff -urN SuiteSparse-7.1.0_orig/Makefile SuiteSparse-7.1.0/Makefile
--- SuiteSparse-7.1.0_orig/Makefile	2023-06-29 19:06:35.000000000 -0700
+++ SuiteSparse-7.1.0/Makefile	2023-07-25 20:17:55.058447922 -0700
@@ -39,8 +39,6 @@
 	( cd SuiteSparse_GPURuntime && $(MAKE) )
 	( cd GPUQREngine && $(MAKE) )
 	( cd SPQR && $(MAKE) )
-	( cd GraphBLAS && $(MAKE) )
-	( cd SPEX && $(MAKE) )
 
 # compile; "make install" only in  SuiteSparse/lib and SuiteSparse/include
 local:
@@ -61,8 +59,6 @@
 	( cd SuiteSparse_GPURuntime && $(MAKE) local )
 	( cd GPUQREngine && $(MAKE) local )
 	( cd SPQR && $(MAKE) local )
-	( cd GraphBLAS && $(MAKE) local )
-	( cd SPEX && $(MAKE) local )
 
 # compile; "sudo make install" will install only in /usr/local
 # (or whatever your CMAKE_INSTALL_PREFIX is)
@@ -84,8 +80,6 @@
 	( cd SuiteSparse_GPURuntime && $(MAKE) global )
 	( cd GPUQREngine && $(MAKE) global )
 	( cd SPQR && $(MAKE) global )
-	( cd GraphBLAS && $(MAKE) global )
-	( cd SPEX && $(MAKE) global )
 
 # install all packages.  Location depends on prior "make", "make global" etc
 install:
@@ -105,8 +99,6 @@
 	( cd SuiteSparse_GPURuntime && $(MAKE) install )
 	( cd GPUQREngine && $(MAKE) install )
 	( cd SPQR && $(MAKE) install )
-	( cd GraphBLAS && $(MAKE) install )
-	( cd SPEX && $(MAKE) install )
 
 # uninstall all packages
 uninstall:
@@ -126,8 +118,6 @@
 	( cd SuiteSparse_GPURuntime && $(MAKE) uninstall )
 	( cd GPUQREngine && $(MAKE) uninstall )
 	( cd SPQR && $(MAKE) uninstall )
-	( cd GraphBLAS && $(MAKE) uninstall )
-	( cd SPEX && $(MAKE) uninstall )
 
 # Remove all files not in the original distribution
 distclean: purge
@@ -154,8 +144,6 @@
 	- $(RM) MATLAB_Tools/*/*.mex* MATLAB_Tools/*/*/*.mex*
 	- $(RM) MATLAB_Tools/*/*.o    MATLAB_Tools/*/*/*.o
 	- $(RM) -r Example/build/*
-	- ( cd GraphBLAS && $(MAKE) purge )
-	- ( cd SPEX && $(MAKE) purge )
 	- $(RM) -r include/* bin/* lib/*
 
 clean: purge
@@ -177,12 +165,9 @@
 	- ( cd CXSparse && $(MAKE) demos )
 	- ( cd RBio && $(MAKE) demos )
 	- ( cd SPQR && $(MAKE) demos )
-	- ( cd GraphBLAS && $(MAKE) demos )
-	- ( cd SPEX && $(MAKE) demos )
 
 # Create the PDF documentation
 docs:
-	( cd GraphBLAS && $(MAKE) docs )
 	( cd Mongoose  && $(MAKE) docs )
 	( cd AMD && $(MAKE) docs )
 	( cd CAMD && $(MAKE) docs )
@@ -191,7 +176,6 @@
 	( cd UMFPACK && $(MAKE) docs )
 	( cd CHOLMOD && $(MAKE) docs )
 	( cd SPQR && $(MAKE) docs )
-	( cd SPEX && $(MAKE) docs )
 
 # statement coverage (Linux only); this requires a lot of time.
 cov: local install
@@ -201,13 +185,6 @@
 	( cd KLU && $(MAKE) cov )
 	( cd SPQR && $(MAKE) cov )
 	( cd UMFPACK && $(MAKE) cov )
-	( cd SPEX && $(MAKE) cov )
-
-gbmatlab:
-	( cd GraphBLAS/GraphBLAS && $(MAKE) )
-
-gblocal:
-	( cd GraphBLAS/GraphBLAS && $(MAKE) local && $(MAKE) install )
 
 debug:
 	( cd SuiteSparse_config && $(MAKE) debug )
@@ -227,6 +204,4 @@
 	( cd SuiteSparse_GPURuntime && $(MAKE) )
 	( cd GPUQREngine && $(MAKE) )
 	( cd SPQR && $(MAKE) debug )
-	( cd GraphBLAS && $(MAKE) debug )
-	( cd SPEX && $(MAKE) debug )
 
diff -urN SuiteSparse-7.1.0_orig/Mongoose/Makefile SuiteSparse-7.1.0/Mongoose/Makefile
--- SuiteSparse-7.1.0_orig/Mongoose/Makefile	2023-06-29 19:06:35.000000000 -0700
+++ SuiteSparse-7.1.0/Mongoose/Makefile	2023-07-25 20:20:39.827860133 -0700
@@ -46,8 +46,8 @@
 	( cd build && cmake $(CMAKE_OPTIONS) -DLOCAL_INSTALL=0 .. && cmake --build . --config Release -j${JOBS} )
 
 # build the Mongoose library (static and dynamic) and run a quick test
-demos: library
-	( cd build ; ./demo )
+# demos: library
+# 	( cd build ; ./demo )
 
 # the same as "make library"
 static: library
