diff -urN SuiteSparse-6.0.3_orig/Makefile SuiteSparse-6.0.3/Makefile
--- SuiteSparse-6.0.3_orig/Makefile	2022-12-24 19:37:26.000000000 -0800
+++ SuiteSparse-6.0.3/Makefile	2023-01-17 09:08:51.599056620 -0800
@@ -39,8 +39,6 @@
 	( cd SuiteSparse_GPURuntime && $(MAKE) )
 	( cd GPUQREngine && $(MAKE) )
 	( cd SPQR && $(MAKE) )
-	( cd GraphBLAS && $(MAKE) )
-	( cd SPEX && $(MAKE) )
 
 # compile; "make install" only in  SuiteSparse/lib and SuiteSparse/include
 local:
@@ -61,8 +59,6 @@
 	( cd SuiteSparse_GPURuntime && $(MAKE) local )
 	( cd GPUQREngine && $(MAKE) local )
 	( cd SPQR && $(MAKE) local )
-	( cd GraphBLAS && $(MAKE) local )
-	( cd SPEX && $(MAKE) local )
 
 # compile; "sudo make install" will install only in /usr/local
 # (or whatever your CMAKE_INSTALL_PREFIX is)
@@ -84,8 +80,6 @@
 	( cd SuiteSparse_GPURuntime && $(MAKE) global )
 	( cd GPUQREngine && $(MAKE) global )
 	( cd SPQR && $(MAKE) global )
-	( cd GraphBLAS && $(MAKE) global )
-	( cd SPEX && $(MAKE) global )
 
 # compile; "sudo make install" will install only in /usr/local
 # (or whatever your CMAKE_INSTALL_PREFIX is)
@@ -107,8 +101,6 @@
 	( cd SuiteSparse_GPURuntime && $(MAKE) both )
 	( cd GPUQREngine && $(MAKE) both )
 	( cd SPQR && $(MAKE) both )
-	( cd GraphBLAS && $(MAKE) both )
-	( cd SPEX && $(MAKE) both )
 
 # install all packages.  Location depends on prior "make", "make global" etc
 install:
@@ -128,8 +120,6 @@
 	( cd SuiteSparse_GPURuntime && $(MAKE) install )
 	( cd GPUQREngine && $(MAKE) install )
 	( cd SPQR && $(MAKE) install )
-	( cd GraphBLAS && $(MAKE) install )
-	( cd SPEX && $(MAKE) install )
 
 # uninstall all packages
 uninstall:
@@ -149,8 +139,6 @@
 	( cd SuiteSparse_GPURuntime && $(MAKE) uninstall )
 	( cd GPUQREngine && $(MAKE) uninstall )
 	( cd SPQR && $(MAKE) uninstall )
-	( cd GraphBLAS && $(MAKE) uninstall )
-	( cd SPEX && $(MAKE) uninstall )
 
 # Remove all files not in the original distribution
 distclean: purge
@@ -177,8 +165,6 @@
 	- $(RM) MATLAB_Tools/*/*.mex* MATLAB_Tools/*/*/*.mex*
 	- $(RM) MATLAB_Tools/*/*.o    MATLAB_Tools/*/*/*.o
 	- $(RM) -r include/* bin/* lib/* Example/build/*
-	- ( cd GraphBLAS && $(MAKE) purge )
-	- ( cd SPEX && $(MAKE) purge )
 
 clean: purge
 
@@ -199,8 +185,6 @@
 	- ( cd CXSparse && $(MAKE) demos )
 	- ( cd RBio && $(MAKE) demos )
 	- ( cd SPQR && $(MAKE) demos )
-	- ( cd GraphBLAS && $(MAKE) demos )
-	- ( cd SPEX && $(MAKE) demos )
 
 # Create the PDF documentation
 docs:
