diff -urN SuiteSparse-5.7.2_orig/Makefile SuiteSparse-5.7.2/Makefile
--- SuiteSparse-5.7.2_orig/Makefile	2020-04-08 13:42:59.000000000 -0700
+++ SuiteSparse-5.7.2/Makefile	2020-05-27 07:26:10.020688473 -0700
@@ -266,6 +266,9 @@
 # hardcoded below.
 include/metis.h:
 ifeq (,$(MY_METIS_LIB))
+	- ( cd metis-5.1.0 && $(MAKE) config prefix=$(SUITESPARSE) cc=$(CC) )
+	- ( cd metis-5.1.0 && $(MAKE) )
+	- ( cd metis-5.1.0 && $(MAKE) install )
 	- ( cd metis-5.1.0 && $(MAKE) config shared=1 prefix=$(SUITESPARSE) cc=$(CC) )
 	- ( cd metis-5.1.0 && $(MAKE) )
 	- ( cd metis-5.1.0 && $(MAKE) install )
